app/src/main/java/com/futurice/freesound/feature/search/SearchFragment.kt

```diff
@@ -0,0 +1,128 @@
+import com.futurice.freesound.feature.common.ui.adapter.RecyclerViewAdapter
+import com.futurice.freesound.inject.fragment.BaseFragmentModule
+import com.futurice.freesound.network.api.model.Sound
+import com.futurice.freesound.viewmodel.DataBinder
+import com.futurice.freesound.viewmodel.ViewModel
+import io.reactivex.disposables.CompositeDisposable
+import kotlinx.android.synthetic.main.fragment_search.*
+import timber.log.Timber
+import javax.inject.Inject
+
+class SearchFragment : BindingBaseFragment<SearchFragmentComponent>() {
+
+    @Inject
+    internal lateinit var searchFragmentViewModel: SearchFragmentViewModel
+
+    @Inject
+    internal lateinit var searchResultAdapter: RecyclerViewAdapter<Sound>
+
+    @Inject
+    internal lateinit var schedulerProvider: SchedulerProvider
+
+    private val dataBinder = object : DataBinder {
+
+        override fun bind(disposables: CompositeDisposable) {
+            disposables += searchFragmentViewModel.soundsOnceAndStream
+                    .subscribeOn(schedulerProvider.computation())
+                    .observeOn(schedulerProvider.ui())
+                    .subscribe({ handleResults(it.orDefault { null }) })
+                    { Timber.e(it, "Error setting Sound items") }
+            disposables += searchFragmentViewModel.searchStateOnceAndStream
+                    .subscribeOn(schedulerProvider.computation())
+                    .observeOn(schedulerProvider.ui())
+                    .subscribe({ showProgress(it) })
+                    { Timber.e(it, "Error receiving search triggered Events") }
+        }
+
+        override fun unbind() {
+            searchFragmentViewModel.stopPlayback()
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater?,
+                              container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? =
+            inflater?.inflate(R.layout.fragment_search, container, false)
+
+    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        recyclerView_searchResults.layoutManager = LinearLayoutManager(activity).apply {
+            recycleChildrenOnDetach = true
+        }
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        recyclerView_searchResults.adapter = searchResultAdapter
+    }
+
+    override fun inject() {
+        component().inject(this)
+    }
+
+    override fun createComponent(): SearchFragmentComponent =
+            (activity as SearchActivity).component()
+                    .plusSearchFragmentComponent(BaseFragmentModule(this))
+
+    override fun viewModel(): ViewModel = searchFragmentViewModel
+
+    override fun dataBinder(): DataBinder = dataBinder
+
+    private fun handleResults(sounds: List<DisplayableItem<Sound>>?) {
+        if (sounds != null) showResults(sounds) else showNothing()
```

Comment:
Didn't you create an extension to handle this situation?

Reply:
I only created `ifNull` extension. Not yet sure if we need one for null and one for not as we can do it with pattern matching